platform/linux-generic/include/odp_ring_internal.h

```diff
@@ -59,7 +56,7 @@ static inline void ring_init(ring_t *ring)
 }
 
 /* Dequeue data from the ring head */
-static inline uint32_t ring_deq(ring_t *ring, uint32_t mask)
+static inline uint32_t ring_deq(ring_t *ring, uint32_t mask, uint32_t *data)
```

Comment:
type might be int here then.

Reply:
The return value usage and type matches with ring_deq_multi() - for both calls it tells how many items were dequeued. It's easier to swap between single and multi calls when function prototypes are similar.

Reply:
Ok.